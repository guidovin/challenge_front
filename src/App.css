.responsive{
  display: flex;
  flex-wrap: wrap;
}
.titleContainer{
  display:flex;
  flex-direction: row;
  justify-content:center;
}
@media (max-width:780px) {
  .titleContainer{
    font-size: 6em !important;
  }
}
@media (max-width:600px) {
  .titleContainer{
    font-size: 4em !important;
  }
}
@media (max-width:340px) {
  .titleContainer{
    font-size: 2em !important;
  }
}
.responsiveContainer{
  background-color: 'grey' !important;
  overflow-y:'scroll' !important;
}
