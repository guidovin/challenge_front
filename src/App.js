import React, { Component } from 'react';
import InterfaceNavbar from './components/interfaceNavbar';
import TodoList from './components/todoList';
import { Container, Header, Segment } from "semantic-ui-react";
import 'semantic-ui-css/semantic.min.css'
import './App.css'
class App extends Component {
  constructor(){
    super();
    this.state = {showCompleted:false, showTodoForm:false , todos:[]};
  };

  togleForm = () => this.setState({showTodoForm: !this.state.showTodoForm});
  togleCompleted = () => this.setState({showCompleted: !this.state.showCompleted});

  updateTodo = (todo) => {
    const newTodo = {
      ...todo,
      completed: !todo.completed,
    };
    fetch('http://localhost:4000/todos/' + todo.userID +'/'+ todo._id,{

      method:'PUT',
      mode:'cors',
      body:JSON.stringify(newTodo),
      headers:{
        'Content-Type': 'application/json'
      }
    })
    .then(response => response.json())
    .then(res => this.setState({todos:res}))
    .catch(error => console.error('Error while updating todo from the browser client:', error));

  }

  deleteTodo = (todo) => {

    fetch('http://localhost:4000/todos/' + todo.userID +'/'+ todo._id,{
      method:'DELETE',
      mode:'cors',
      headers:{
        'Content-Type': 'application/json'
      }
    })
    .then(response => response.json())
    .then(res => this.setState({todos:res}))
    .catch(error => console.error('Error while trying to delete todo from the browser client:', error));

  }

  createTodo = (todo) => {
    console.log(todo.text);
    const newTodo = {
      text: todo.text,
    };
    fetch('http://localhost:4000/todos/'+todo.userID,{
      method:'POST',
      mode:'cors',
      body:JSON.stringify(newTodo),
      headers:{
        'Content-Type': 'application/json'
      }
    })
    .then(response => response.json())
    .then(res => this.setState({todos:res}))
    .catch(error => console.error('Error while trying to create todo from the browser client:', error));
  }

  render() {
    return (
      <Container style={{ margin: 20 , width:'75vw'}}>

        <Segment className="responsiveContainer" style={{ backgroundColor: 'rgb(227, 234, 249)', overflowY:'scroll'}}>
          <div className="titleContainer" style={{fontFamily:'Mosk', fontSize:'8em', height:'1.8em', color:'rgb(247, 180, 128)'}} >
            <p>To do app</p>
          </div>
          <InterfaceNavbar
            showCompleted= {this.state.showCompleted}
            togleCompleted= {this.togleCompleted.bind(this)}
            togleForm= {this.togleForm.bind(this)}
            createTodo= {this.createTodo.bind(this)}
          />
          <TodoList
            showCompleted= {this.state.showCompleted}
            todos= {this.state.todos}
            updateTodo= {this.updateTodo.bind(this)}
            deleteTodo= {this.deleteTodo.bind(this)}
          />
        </Segment>

      </Container>
    );
  }

  componentDidMount() {
    let userID = window.location.pathname.split('/')[2];
    fetch('http://localhost:4000/todos/'+userID)
    .then(res => {return res.json()})
    .then(todoList =>{this.setState({todos:todoList}); })
    .catch((err, res)=>{} )
  }
}
export default App;
