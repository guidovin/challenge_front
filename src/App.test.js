import React from 'react';
import { mount } from 'enzyme';
import ReactDOM from 'react-dom';
import App from './App.js';

describe('App', () => {
  it('should render correctly, have children, pass functions to them and alter state correctly ', () => {
    const component = mount(<App />);
    component.find('.viewTogle').hostNodes().simulate('click');
    expect(component.state('showCompleted')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
