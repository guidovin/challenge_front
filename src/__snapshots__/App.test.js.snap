// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly and have children 1`] = `
<App>
  <Container
    style={
      Object {
        "margin": 20,
        "width": "75vw",
      }
    }
  >
    <div
      className="ui container"
      style={
        Object {
          "margin": 20,
          "width": "75vw",
        }
      }
    >
      <Segment
        className="responsiveContainer"
        style={
          Object {
            "backgroundColor": "rgb(227, 234, 249)",
            "overflowY": "scroll",
          }
        }
      >
        <div
          className="ui segment responsiveContainer"
          style={
            Object {
              "backgroundColor": "rgb(227, 234, 249)",
              "overflowY": "scroll",
            }
          }
        >
          <div
            className="titleContainer"
            style={
              Object {
                "color": "rgb(247, 180, 128)",
                "fontFamily": "Mosk",
                "fontSize": "8em",
                "height": "1.8em",
              }
            }
          >
            <p>
              To do app
            </p>
          </div>
          <InterfaceNavbar
            createTodo={[Function]}
            showCompleted={true}
            togleCompleted={[Function]}
            togleForm={[Function]}
          >
            <Menu>
              <div
                className="ui menu"
              >
                <MenuItem
                  active={false}
                  className="viewTogle"
                  name="showCompleted"
                  onClick={[Function]}
                  position="left"
                >
                  <a
                    className="left item viewTogle"
                    onClick={[Function]}
                  >
                    Showing all
                  </a>
                </MenuItem>
                <MenuMenu
                  className="todoInput"
                  position="right"
                >
                  <div
                    className="right menu todoInput"
                  >
                    <div
                      className="ui right aligned category search item todoInput"
                    >
                      <div
                        className="ui transparent icon input todoInput"
                      >
                        <input
                          className="prompt"
                          id="inputText"
                          name="todoText"
                          onChange={[Function]}
                          onKeyUp={[Function]}
                          placeholder="Todo name..."
                          type="text"
                        />
                      </div>
                    </div>
                    <i
                      className="right arrow link icon submit"
                      onClick={[Function]}
                    />
                  </div>
                </MenuMenu>
              </div>
            </Menu>
          </InterfaceNavbar>
          <TodoList
            deleteTodo={[Function]}
            showCompleted={true}
            todos={Array []}
            updateTodo={[Function]}
          >
            <div
              className="responsiveContainer"
            />
          </TodoList>
        </div>
      </Segment>
    </div>
  </Container>
</App>
`;

exports[`App should render correctly, have children, pass functions to them and alter state correctly  1`] = `
<App>
  <Container
    style={
      Object {
        "margin": 20,
        "width": "75vw",
      }
    }
  >
    <div
      className="ui container"
      style={
        Object {
          "margin": 20,
          "width": "75vw",
        }
      }
    >
      <Segment
        className="responsiveContainer"
        style={
          Object {
            "backgroundColor": "rgb(227, 234, 249)",
            "overflowY": "scroll",
          }
        }
      >
        <div
          className="ui segment responsiveContainer"
          style={
            Object {
              "backgroundColor": "rgb(227, 234, 249)",
              "overflowY": "scroll",
            }
          }
        >
          <div
            className="titleContainer"
            style={
              Object {
                "color": "rgb(247, 180, 128)",
                "fontFamily": "Mosk",
                "fontSize": "8em",
                "height": "1.8em",
              }
            }
          >
            <p>
              To do app
            </p>
          </div>
          <InterfaceNavbar
            createTodo={[Function]}
            showCompleted={true}
            togleCompleted={[Function]}
            togleForm={[Function]}
          >
            <Menu>
              <div
                className="ui menu"
              >
                <MenuItem
                  active={false}
                  className="viewTogle"
                  name="showCompleted"
                  onClick={[Function]}
                  position="left"
                >
                  <a
                    className="left item viewTogle"
                    onClick={[Function]}
                  >
                    Showing all
                  </a>
                </MenuItem>
                <MenuMenu
                  className="todoInput"
                  position="right"
                >
                  <div
                    className="right menu todoInput"
                  >
                    <div
                      className="ui right aligned category search item todoInput"
                    >
                      <div
                        className="ui transparent icon input todoInput"
                      >
                        <input
                          className="prompt"
                          id="inputText"
                          name="todoText"
                          onChange={[Function]}
                          onKeyUp={[Function]}
                          placeholder="Todo name..."
                          type="text"
                        />
                      </div>
                    </div>
                    <i
                      className="right arrow link icon submit"
                      onClick={[Function]}
                    />
                  </div>
                </MenuMenu>
              </div>
            </Menu>
          </InterfaceNavbar>
          <TodoList
            deleteTodo={[Function]}
            showCompleted={true}
            todos={Array []}
            updateTodo={[Function]}
          >
            <div
              className="responsiveContainer"
            />
          </TodoList>
        </div>
      </Segment>
    </div>
  </Container>
</App>
`;
